src/sentry/static/sentry/app/views/projectGeneralSettings.jsx

```diff
@@ -134,17 +190,63 @@ export default class ProjectGeneralSettings extends AsyncView {
       );
     } else {
       return (
-        <p>
-          <a
-            href={`/${orgId}/${projectId}/settings/transfer/`}
-            className="btn btn-danger pull-right"
+        <Field
+          label={tct('Transfer the [project] project and all related data.', {
+            project: <strong>{project.slug}</strong>,
+          })}
+          help={t('Careful, this action cannot be undone.')}
+        >
+          <Confirm
+            onConfirm={this.handleTransferProject}
+            priority="danger"
+            title={`${t('Transfer project')}?`}
+            confirmText={t('Transfer project')}
+            renderMessage={({doConfirm, doClose}) => (
+              <div>
+                <TextBlock>
+                  <strong>
+                    {t('Transferring this project is permanent and cannot be undone!')}
+                  </strong>
+                </TextBlock>
+                <TextBlock>
+                  {t(
+                    'Please enter the owner of the organization you would like to transfer this project to.'
+                  )}
+                </TextBlock>
+                <Panel>
+                  <PanelHeader>{t('Transfer to')}</PanelHeader>
+                  <Form
+                    hideFooter
+                    onFieldChange={this.handleTransferFieldChange}
+                    onSubmit={(data, onSuccess, onError, e) => {
+                      e.stopPropagation();
+                      doConfirm();
+                    }}
+                  >
+                    <NewTextField
+                      name="email"
+                      label={t('Organization Owner')}
+                      placeholder="admin@example.com"
+                      required
+                      help={tct(
+                        'A request will be emailed to the new owner in order to transfer [project] to a new organization.',
+                        {
+                          project: <strong> {project.slug} </strong>,
+                        }
+                      )}
+                    />
+                  </Form>
+                </Panel>
+              </div>
+            )}
           >
-            {t('Transfer Project')}
-          </a>
-          Transfer the <strong>{project.slug}</strong> project and all related data.
-          <br />
-          Careful, this action cannot be undone.
-        </p>
+            <div>
+              <Button className="ref-transfer-project" type="button" priority="danger">
```

Comment:
Could we avoid classnames just for tests? I know it can be harder since we can't always use stuff like `:contains`, but would prefer something like `data-test="remove-btn"` that explicitly marks it as a thing that's used for tests only.

Reply:
We use `ref-` class names to signify that its only used for tests. Let's leave this for now, but its worth revisiting how we want to handle this and do a bulk refactor.